prow/plugins/label/label_test.go

```diff
@@ -715,3 +691,88 @@ func TestRepeat(t *testing.T) {
+func TestSlackMessages(t *testing.T) {
+
+	type testCase struct {
+		name                   string
+		body                   string
+		expectedMessages       map[string][]string
+		expectedRepeatedLabels []string
+		issueLabels            []string
+		repoLabels             []string
+		commenter              string
+	}
+	testcases := []testCase{
+		{
+			name:             "If sig mentioned then we send a message to the sig with the body of the comment",
+			body:             "@kubernetes/sig-node-misc This issue needs update.",
+			expectedMessages: map[string][]string{"sig-node": {"This issue needs update."}},
+			repoLabels:       []string{"sig/node", "sig/api-machinery"},
+			issueLabels:      []string{},
+			commenter:        orgMember,
+		},
+		{
+			name:             "If multiple sigs mentioned, we send a message to each sig with the body of the comment",
+			body:             "@kubernetes/sig-node-misc, @kubernetes/sig-testing-misc Message sent to multiple sigs.",
+			expectedMessages: map[string][]string{"sig-testing": {"Message sent to multiple sigs."}, "sig-node": {"Message sent to multiple sigs."}},
+			repoLabels:       []string{"sig/node", "sig/api-machinery"},
+			issueLabels:      []string{},
+			commenter:        orgMember,
+		},
+		{
+			name:             "Message should not be sent if the pattern for the channel does not match",
+			body:             "@kubernetes/node-misc No message sent",
+			expectedMessages: map[string][]string{},
+			repoLabels:       []string{"sig/node", "sig/api-machinery"},
+			issueLabels:      []string{},
+			commenter:        orgMember,
+		},
+		{
+			name:             "Message sent only if the pattern for the channel match",
+			body:             "@kubernetes/node-misc @kubernetes/sig-api-machinery-bugs Message sent to matching sigs.",
+			expectedMessages: map[string][]string{"sig-api-machinery": {"Message sent to matching sigs."}},
+			repoLabels:       []string{"sig/node", "sig/api-machinery"},
+			issueLabels:      []string{},
+			commenter:        orgMember,
+		},
+	}
+
+	//Test that this functionality works on comments, newly opened issues and newly opened PRs
+	fakeRepoFunctions := []func(string, string, []string, []string) (*fakegithub.FakeClient, assignEvent){
+		getFakeRepoIssueComment,
+		getFakeRepoIssue,
+		getFakeRepoPullRequest,
+	}
+
+	for _, tc := range testcases {
+		sort.Strings(tc.expectedRepeatedLabels)
+
+		for i := 0; i < len(fakeRepoFunctions); i++ {
+			fakeClient, ae := fakeRepoFunctions[i](tc.body, tc.commenter, tc.repoLabels, tc.issueLabels)
+			fakeSlackClient := &fakeslack.FakeClient{
+				SentMessages: make(map[string][]string),
+			}
+
+			if err := handle(fakeClient, logrus.WithField("plugin", pluginName), ae, fakeSlackClient); err != nil {
+				t.Errorf("For case %s, didn't expect error from label test: %v", tc.name, err)
+				return
+			}
+			if len(tc.expectedMessages) != len(fakeSlackClient.SentMessages) {
+				t.Errorf("The number of messages sent do not tally.")
```

Comment:
`t.Fatalf` and remove the `return`. Also make the error message more helpful, such as `got %d, expected %d`.